### PR TITLE
Embed SignedCommon in Root, Snapshot, and Timestamp

### PR DESCRIPTION
**Pull request by cyli**
Since it's already embedded in `Targets`.

Will be useful for me for https://github.com/docker/notary/pull/580

### CODE REVIEW

File: tuf/data/root_test.go [hunk 1]
```diff
@@ -52,7 +54,7 @@ func validRootTemplate() *SignedRoot {
 }
 
 func TestRootToSignedMarshalsSignedPortionWithCanonicalJSON(t *testing.T) {
-	r := SignedRoot{Signed: Root{Type: "root", Version: 2, Expires: time.Now()}}
+	r := SignedRoot{Signed: Root{SignedCommon: SignedCommon{Type: "root", Version: 2, Expires: time.Now()}}}
```
Comment (riyazdf) [inline tuf/data/root_test.go]: nit: should this be uppercase `"Root"`?  It shouldn't affect the behavior of this test but it seems like we should keep this consistent with what is expected (there are a few other instances in this file)

File: tuf/data/timestamp.go [hunk 2]
```diff
@@ -64,9 +61,11 @@ func NewTimestamp(snapshot *Signed) (*SignedTimestamp, error) {
 	return &SignedTimestamp{
 		Signatures: make([]Signature, 0),
 		Signed: Timestamp{
-			Type:    TUFTypes["timestamp"],
-			Version: 0,
-			Expires: DefaultExpires("timestamp"),
+			SignedCommon: SignedCommon{
+				Type:    TUFTypes["timestamp"],
```
Comment (riyazdf) [inline tuf/data/timestamp.go]: nit: can we use the `CanonicalTimestampRole` constant here and below for `Expires:`?

**Comment by riyazdf** [comment]
LGTM after comments!

File: tuf/data/root_test.go (hunk 1 above)
Comment (cyli) [inline tuf/data/root_test.go]: Yep, thanks for noticing that!

File: tuf/data/timestamp.go (hunk 2 above)
Comment (cyli) [inline tuf/data/timestamp.go]: Yes, good idea.

**Comment by endophage** [comment]
LGTM!

**Pull request merged**